src/hooks/usePublishCoordinator.js

```diff
@@ -0,0 +1,89 @@
+import partititon from 'lodash/partition';
+import {
+  useCallback,
+  useRef
+} from 'react';
+
+import {
+  useOkapiKy,
+  useStripes,
+} from '@folio/stripes/core';
+
+export const PUBLISH_COORDINATOR_STATUSES = {
+  COMPLETE: 'COMPLETE',
+  ERROR: 'ERROR',
+  IN_PROGRESS: 'IN_PROGRESS',
+};
+
+export const PUBLICATIONS_API = 'publications';
+export const CONSORTIA_API = 'consortia';
+
+
+export const TIMEOUT = 2500;
+
+const formatPublicationResult = ({ publicationResults, totalRecords }) => {
+  const [results, errors] = partititon(publicationResults, ({ statusCode }) => statusCode >= 200 && statusCode < 300);
+
+  const formattedResults = results.map(({ response, ...rest }) => ({
+    response: JSON.parse(response),
+    ...rest,
+  }));
+
+  return {
+    publicationResults: formattedResults,
+    publicationErrors: errors,
+    totalRecords,
+  };
+};
+
+export const usePublishCoordinator = (namespace, options = {}) => {
+  const ky = useOkapiKy();
+  const stripes = useStripes();
+  const abortController = useRef(new AbortController());
+
+  const consortium = stripes.user?.user?.consortium;
+  const baseApi = `${CONSORTIA_API}/${consortium?.id}/${PUBLICATIONS_API}`;
+
+  const getPublicationResults = useCallback((id, { signal }) => {
+    return ky.get(`${baseApi}/${id}/results`, { signal })
+      .json()
+      .then(formatPublicationResult);
+  }, [ky, baseApi]);
+
+  const getPublicationDetails = useCallback(async (requestId, { signal } = {}) => {
+    const { id, status } = await ky.get(`${baseApi}/${requestId}`, { signal }).json();
+
+    if (status !== PUBLISH_COORDINATOR_STATUSES.IN_PROGRESS) return getPublicationResults(id, { signal });
+
+    await new Promise((resolve) => setTimeout(resolve, TIMEOUT));
```

Comment:
Why is there a need for 2.5 seconds timeout?

Reply:
That implementation was taken from ui-consortia-settings. https://github.com/folio-org/ui-consortia-settings/blob/master/src/hooks/usePublishCoordinator/usePublishCoordinator.js

Reply:
In this case you can import this hook from `stripes-acq-components`
https://github.com/folio-org/stripes-acq-components/blob/master/lib/hooks/consortia/usePublishCoordinator/usePublishCoordinator.js